hadoop-crypto/src/main/java/com/palantir/hadoop/EncryptedFileSystem.java

```diff
@@ -76,6 +79,29 @@ public FSDataOutputStream create(Path path, FsPermission permission,
         return os;
     }
 
+    @Override
+    public boolean rename(Path src, Path dst) throws IOException {
+        KeyMaterial keyMaterial = keyStore.get(src.toString());
+        keyStore.put(dst.toString(), keyMaterial);
+        boolean renamed = fs.rename(src, dst);
+
+        if (renamed) {
+            tryRemoveKey(src);
+        } else {
+            tryRemoveKey(dst);
+        }
+
+        return renamed;
+    }
+
+    private void tryRemoveKey(Path path) {
+        try {
+            keyStore.remove(path.toString());
+        } catch (Exception e) {
+            log.warn("Unable to remove KeyMaterial for file: {}", path);
```

Comment:
don't think that's right. `tryRemoveKey` is supposed to be best effort removal. The signal for failure to rename is the boolean returned by the method.